common/msg_queue.c

```diff
@@ -9,11 +10,37 @@ struct msg_queue {
 	const u8 **q;
 };
 
+static int extract_fd(const u8 *msg)
+{
+	const u8 *p = msg + sizeof(u16);
+	size_t len = tal_count(msg) - sizeof(u16);
+
+	if (fromwire_peektype(msg) != MSG_PASS_FD)
+		return -1;
+
+	return fromwire_u32(&p, &len);
+}
+
+/* Close any fds left in queue! */
+static void destroy_msg_queue(struct msg_queue *q)
+{
+	for (size_t i = 0; i < tal_count(q->q); i++) {
+		int fd = extract_fd(q->q[i]);
+		if (fd != -1) {
+			close(fd);
+			abort();
```

Comment:
what's the rationale for calling `abort()` here?

Reply:
Oops,debugging check!!!  Wanted to see if it was hit, good catch!